charts/telegraf/templates/_helpers.tpl

```diff
@@ -201,29 +201,49 @@ app.kubernetes.io/instance: {{ .Release.Name }}
       {{ $key }} = {{ $value }}
           {{- end }}
           {{- if eq $tp "[]interface {}" }}
-      {{ $key }} = [
-              {{- $numOut := len $value }}
-              {{- $numOut := sub $numOut 1 }}
+            {{- $val0 := index $value 0 -}}
+            {{- $val0tp := typeOf $val0 -}}
+            {{- if eq $val0tp "map[string]interface {}" -}}
```

Comment:
Maybe when the variables are not used afterwards, this could become one-liner?
```suggestion
            {{- if eq (index $value 0 | typeOf) "map[string]interface {}" -}}
```